### PR TITLE
Reset mongo client

### PR DESCRIPTION
**Pull request by b0b3rt**
Quick & dirty patch to see if reassigning the Mongo client every 10 minutes mitigates the memory leak (since current best guess is that something in the client gets into a bad state and begins to retain all the response info).

Not sure about a couple things here:

1. Whether we do in fact want to use the `returnNonCachedInstance` flag
1. Whether we should be attempting to close the existing client either before or after spinning a new one up

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1202937135416915) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/server/serverStartup.ts [hunk 1]
```diff
@@ -31,6 +57,17 @@ async function serverStartup() {
...
+    setInterval(async () => {
+      const memoryUsage = process.memoryUsage()?.heapTotal;
+      if (memoryUsage > consoleLogMemoryUsageThreshold.get() * 2) {
+        await setMongoClient(commandLineArguments.mongoUrl);
+      }
+    }, 10000);
+  });
   
   try {
     // eslint-disable-next-line no-console
```
Comment (jpaddison3) [inline packages/lesswrong/server/serverStartup.ts:73]: Seems like you should use `setMongoClient` here?

File: packages/lesswrong/server/serverStartup.ts [hunk 2]
```diff
@@ -31,6 +57,17 @@ async function serverStartup() {
   });
   
   const commandLineArguments = getCommandLineArguments();
+
+  await setMongoClient(commandLineArguments.mongoUrl);
+
+  onStartup(() => {
+    setInterval(async () => {
+      const memoryUsage = process.memoryUsage()?.heapTotal;
+      if (memoryUsage > consoleLogMemoryUsageThreshold.get() * 2) {
```
Comment (jpaddison3) [inline packages/lesswrong/server/serverStartup.ts:66]: Yikes. I don't love reusing a setting whose name tells me it's purely for logging for controlling somewhat risky behavior.

File: packages/lesswrong/server/serverStartup.ts [hunk 3]
```diff
@@ -31,6 +57,17 @@ async function serverStartup() {
...
+        await setMongoClient(commandLineArguments.mongoUrl);
```
Comment (jpaddison3) [inline packages/lesswrong/server/serverStartup.ts:67]: I want a console log here.

**Comment by jpaddison3** [comment]
I tried running this locally and got

```
MongoError: Topology is closed, please connect
```

As a permanent error.

**Comment by darkruby501** [comment]
Fixed the real cause of memory leak

**Pull request closed without merging**